tests/user_guide_test.py

```diff
@@ -22,6 +45,130 @@ def user_guide_snippet_dynamic_scratch_var() -> Expr:
     )
 
 
+# RECURSIVE VERSION HAS AN INFINITE LOOP!!!!
+@Subroutine(TealType.uint64, input_types=[TealType.uint64, TealType.uint64])
```

Comment:
Is this a real bug?

Reply:
Shouldn't `buggy_euclid(x, Mod(y, x))` be `buggy_euclid(y, Mod(x, y))`?

Reply:
No it's not a real bug. Upshot, the python equivlent had the same bug. The fix is the following
(basically swap x and y in the `Else`):


```python
@Subroutine(TealType.uint64, input_types=[TealType.uint64, TealType.uint64])
def buggy_euclid(x, y):
    return (
        If(x < y)
        .Then(buggy_euclid(y, x))
        .Else(If(y == Int(0)).Then(x).Else(buggy_euclid(y, Mod(x, y))))
    )
```

# TODO: run the blackbox agsint this new "correction"

## Python model to assert as well:
```python
def be(x, y):
    return be(y, x) if x < y else (x if y == 0 else be(y, x % y))
```